### PR TITLE
Peak comparison tool updates

### LINKED ISSUE #18043: Peaks comparison tool modifications
Some feedback from Pascal.

 - [ ] ~Need to add mechanism to handle multiple peaks in spectrum.~
    - Angle is not dependant on TOF/ d-spacing?
 - [ ] ~Would be nice to add a peak if none exists.~
    - Overcomplicates tool, what if you want to go back and reselect peaks?
 - [x] Bug with line being left if you change to another tool.
 - [ ] ~Direction angles not correct~. 
    - See #18120
 - [ ] We should also add an entry for this into the wiki pages.

### PR DESCRIPTION
**Pull request by samueljackson92**
Fixed two issues:
 - Swapped to calculating the angle between Q vectors
 - Fixed bug where line would be left after changing between tools or clearing/erasing peaks 

**To test:**

 - Load some single crystal data
 - Open the instrument view and choose some peaks
 - Select the peak comparison tool
 - Choose two peaks and check the following
    - the correct reciprocal angle between the two Q vectors is calculated
 - Clear the peaks.
    - the line between two peaks should be removed

Fixes #18043.

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####

- [x] Is the code of an acceptable quality?
- [x] Does the code conform to the [coding standards](http://www.mantidproject.org/Coding_Standards)? Is it well structured with small focussed classes/methods/functions?
- [ ] Are there unit/system tests in place? Are the unit tests small and test the a class in isolation?
- [ ] If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- [x] Do changes function as described? Add comments below that describe the tests performed?
- [ ] How do the changes handle unexpected situations, e.g. bad input?
- [ ] Has the relevant documentation been added/updated?
- [ ] Is user-facing documentation written in a user-friendly manner?
- [ ] Has developer documentation been updated if required?

- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: MantidQt/MantidWidgets/src/InstrumentView/InstrumentWidgetPickTab.cpp [hunk 1]
```diff
@@ -545,11 +545,11 @@ void InstrumentWidgetPickTab::initSurface() {
...
   connect(
       surface,
-      SIGNAL(comparePeaks(
-          std::pair<Mantid::Geometry::IPeak *, Mantid::Geometry::IPeak *>)),
+      SIGNAL(comparePeaks(std::pair<std::vector<Mantid::Geometry::IPeak *>,
+                                    std::vector<Mantid::Geometry::IPeak *>>)),
       this,
-      SLOT(comparePeaks(
-          std::pair<Mantid::Geometry::IPeak *, Mantid::Geometry::IPeak *>)));
+      SLOT(comparePeaks(std::pair<std::vector<Mantid::Geometry::IPeak *>,
```
Comment (AntonPiccardoSelg) [inline MantidQt/MantidWidgets/src/InstrumentView/InstrumentWidgetPickTab.cpp]: Wouldn't you need the `const &` qualifier here?

File: MantidQt/MantidWidgets/src/InstrumentView/ProjectionSurface.cpp [hunk 2]
```diff
@@ -738,47 +741,67 @@ void ProjectionSurface::touchComponentAt(int x, int y) {
 void ProjectionSurface::erasePeaks(const QRect &rect) {
   foreach (PeakOverlay *po, m_peakShapes) {
     po->selectIn(rect);
+    auto peakMarkers = po->getSelectedPeakMarkers();
+
+    // clear selected peak markers
+    for (auto marker : peakMarkers) {
```
Comment (AntonPiccardoSelg) [inline MantidQt/MantidWidgets/src/InstrumentView/ProjectionSurface.cpp]: Did you want `auto&`?

File: MantidQt/MantidWidgets/src/InstrumentView/ProjectionSurface.cpp [hunk 3]
```diff
@@ -738,47 +741,67 @@ void ProjectionSurface::touchComponentAt(int x, int y) {
...
-      peak = po->getPeaksWorkspace()->getPeakPtr(marker->getRow());
-      break;
+
+    // make the assumption that the first peak found in the recticule is the one
+    // we wanted.
+    if (markers.length() > 0 && origin.isNull()) {
+      origin = markers.first()->origin();
+    }
+
+    for (auto marker : markers) {
```
Comment (AntonPiccardoSelg) [inline MantidQt/MantidWidgets/src/InstrumentView/ProjectionSurface.cpp]: Did you want `auto&`?

File: MantidQt/MantidWidgets/src/InstrumentView/ProjectionSurface.cpp [hunk 4]
```diff
@@ -738,47 +741,67 @@ void ProjectionSurface::touchComponentAt(int x, int y) {
...
+  for (auto *po : m_peakShapes) {
     po->selectIn(rect);
     const auto markers = po->getSelectedPeakMarkers();
-    if (markers.length() > 0) {
-      marker = markers.first();
-      origin = marker->origin();
-      peak = po->getPeaksWorkspace()->getPeakPtr(marker->getRow());
-      break;
+
+    // make the assumption that the first peak found in the recticule is the one
```
Comment (AntonPiccardoSelg) [inline MantidQt/MantidWidgets/src/InstrumentView/ProjectionSurface.cpp]: Typo?

File: MantidQt/MantidWidgets/src/InstrumentView/ProjectionSurface.cpp (hunk 3 above)
Comment (samueljackson92) [inline MantidQt/MantidWidgets/src/InstrumentView/ProjectionSurface.cpp]: Agreed.

File: MantidQt/MantidWidgets/src/InstrumentView/ProjectionSurface.cpp (hunk 2 above)
Comment (samueljackson92) [inline MantidQt/MantidWidgets/src/InstrumentView/ProjectionSurface.cpp]: Agreed.

File: MantidQt/MantidWidgets/src/InstrumentView/InstrumentWidgetPickTab.cpp (hunk 1 above)
Comment (samueljackson92) [inline MantidQt/MantidWidgets/src/InstrumentView/InstrumentWidgetPickTab.cpp]: Yes.

File: MantidQt/MantidWidgets/src/InstrumentView/ProjectionSurface.cpp (hunk 4 above)
Comment (samueljackson92) [inline MantidQt/MantidWidgets/src/InstrumentView/ProjectionSurface.cpp]: Yes.

**Comment by AntonPiccardoSelg** [comment]
All looks good, checked the angle and it is correct. 

As already discussed, we should get rid of the connection line in the SliceViewer, shifting the time-of-flight slider and there are no peaks present.

**Comment by AntonPiccardoSelg** [comment]
The line now vanishes when the time slider is moved as one would expect. Please :shipit:

**Pull request merged**